### PR TITLE
feat: add reference file schema option for federated formats

### PR DESCRIPTION
**Pull request by Neenu1995**

### CODE REVIEW

File: google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java [hunk 1]
```diff
@@ -4586,4 +4587,217 @@ public void testPreserveAsciiControlCharacters()
...
+
+      LoadJobConfiguration loadJobConfiguration =
+          LoadJobConfiguration.newBuilder(tableId, SOURCE_URIS, FormatOptions.avro())
+              .setReferenceFileSchemaUri(referenceFileSchema)
+              .build();
+
+      Job job = bigquery.create(JobInfo.of(loadJobConfiguration));
+      // Blocks until this load table job completes its execution, either failing or succeeding.
+      job = job.waitFor();
+      if (job.isDone()) {
```
Comment (shollyman) [inline google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java:4637]: this logic seems odd (here and in the other variants).  A job can be done with or without an error attached.  you can just assert the job completed without errors?

Also, as this doesn't appear to be a sample it's not clear what emitting to system.out is getting you.

File: google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/ExternalTableDefinition.java [hunk 2]
```diff
@@ -317,6 +328,15 @@ com.google.api.services.bigquery.model.ExternalDataConfiguration toExternalDataC
     if (getAutodetect() != null) {
       externalConfigurationPb.setAutodetect(getAutodetect());
     }
+    if (FormatOptions.AVRO.equals(getFormatOptions().getType())
```
[the remark this answers is not in the document]
Comment (Neenu1995) [reply google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/ExternalTableDefinition.java]: done

File: google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java (hunk 1 above)
Comment (Neenu1995) [reply google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java]: fixed

**Review by shollyman — APPROVED** [review]
Thanks!

**Pull request merged**